### DISCUSSION TITLE
Access abandoned Talos cluster

### DISCUSSION THREAD

**Discussion by mathrock74** [siderolabs/talos#9445] [Q&A] [answered]
Hi,

my workmate created a 3 node Talos cluster in VMs that I gave him. He abruptly left our institution without leaving any docs behind. I would like to access his Talos cluster now. I can access the VM consoles and I have admin access to the k8s that's running on Talos. I tried to use talosctl but I only get "error constructing client: failed to determine endpoints" for every command. Can someone please advise what to do?

thank you

**Comment by smira** [comment, marked as answer]
If you have admin-level kubeconfig, you can try the following trick:

```
kubectl debug -it node/<controlpalne-node-name>  --profile=sysadmin -n kube-system --image=alpine
```

With that, recover controlplane machine config from `/host/system/state/config.yaml`.

After that, follow https://www.talos.dev/v1.8/talos-guides/howto/cert-management/#from-control-plane-machine-configuration to recover `talosconfig`.

**Reply by mathrock74** [reply]
Thanks a lot! Although I have almost no clue what I'm doing this actually worked (had to use "--profile=general") and I was able to shutdown the talos nodes in an orderly manner (instead of just killing the VMs). It was more of an intellectual exercise, nothing related to production work.
